controller/controller.go

```diff
@@ -53,7 +54,7 @@ type Controller struct {
 	isShutdown concurrenz.AtomicBoolean
 }
 
-func NewController(cfg *Config) (*Controller, error) {
+func NewController(cfg *Config, versionProvider common.VersionProvider) (*Controller, error) {
```

Comment:
VersionProvider allows us to invert who is providing the build version information (i.e. the `ziti` repo)

Reply:
Any reason why we wouldn't just have `NewController` just do `NewVersionProviderImpl` itself and call it a day? Not sure we _really_ need to pass that in?

Reply:
The provider has build values such as version, revision, etc. that are set externally. If we pulled from foundation none of those values would be available.